src/offline.cpp

```diff
@@ -23,10 +23,50 @@ Q_DECLARE_LOGGING_CATEGORY(PACKAGEKITQT_OFFLINE)
 
 using namespace PackageKit;
 
+Offline::Results::Results(const QDBusPendingCall &call)
+    : m_reply(call)
+{
+    m_reply.waitForFinished();
```

Comment:
If we do this,  then we have to block. It could be useful only to wait when the results are requested.

Reply:
You mean, add `Results::waitForFinished()` or rather call `m_reply.waitForFinished()` in each getter?